flocker/ca/_validation.py

```diff
@@ -130,11 +132,16 @@ def treq_with_authentication(reactor, ca_path, user_cert_path, user_key_path):
     :param FilePath ca_path: Absolute path to the public cluster certificate.
     :param FilePath user_cert_path: Absolute path to the user certificate.
     :param FilePath user_key_path: Absolute path to the user private key.
+    :param bool persistent_connections: Whether to use persistent connections
+        in the treq client that is created.
 
     :return: ``treq`` compatible object.
     """
     ca = Certificate.loadPEM(ca_path.getContent())
     user_credential = UserCredential.from_files(user_cert_path, user_key_path)
     policy = ControlServicePolicy(
         ca_certificate=ca, client_credential=user_credential.credential)
-    return HTTPClient(Agent(reactor, contextFactory=policy))
+    pool = None
+    if persistent_connections:
+        pool = HTTPConnectionPool(reactor, persistent_connections)
```

Comment:
:question: Perhaps we should always supply a connection pool, persistent or not.